crates/legacy/src/service.rs

```diff
@@ -181,6 +184,9 @@ pub struct GlobalState<Types: NodeType> {
 
     pub block_size_limits: BlockSizeLimits,
 
+    // A mapping from transaction hash to its status
+    pub tx_status: HashMap<Commitment<Types::Transaction>, TransactionStatus>,
```

Comment:
We need some mechanism to prune this map, using something like `RotatingSet` or `Lru` here, otherwise we'll OOM eventually.

Reply:
This is exactly the next thing I'm going to add. Originally I thought it will be another wrapper and needs lots of updates, but now think about it, changing to `RotatingSet` sounds like a straightforward way.

Reply:
Oh, sorry, I didn't notice that you've already included that in PR description

Reply:
Updated in [aacb434](https://github.com/EspressoSystems/marketplace-builder-core/pull/236/commits/aacb4345c61cb046785da95fc7c3a9dfae742920), this is just easy pruning, will add more like "set a minimum retention" and "set a target retention" in the issue description when I get a chance.